go-controller/pkg/node/gateway_localnet_linux_test.go

```diff
@@ -525,6 +525,68 @@ var _ = Describe("Node Operations", func() {
 			err := app.Run([]string{app.Name})
 			Expect(err).NotTo(HaveOccurred())
 		})
+
+		It("inits iptables rules for ExternalIP with DualStack", func() {
+			app.Action = func(ctx *cli.Context) error {
+
+				externalIPv4 := "10.10.10.1"
+				externalIPv6 := "fd00:96:1::1"
+				clusterIPv4 := "10.129.0.2"
+				clusterIPv6 := "fd00:10:96::10"
+				fNPW.gatewayIPv6 = v6localnetGatewayIP
+
+				service := *newService("service1", "namespace1", clusterIPv4,
+					[]v1.ServicePort{
+						{
+							Port:     8032,
+							Protocol: v1.ProtocolTCP,
+						},
+					},
+					v1.ServiceTypeClusterIP,
+					[]string{externalIPv4, externalIPv6},
+				)
+				service.Spec.ClusterIPs = []string{clusterIPv4, clusterIPv6}
+				fNPW.addService(&service)
+
+				expectedTables4 := map[string]util.FakeTable{
+					"filter": {
+						"OVN-KUBE-EXTERNALIP": []string{
+							fmt.Sprintf("-p %s -d %s --dport %v -j ACCEPT", service.Spec.Ports[0].Protocol, externalIPv4, service.Spec.Ports[0].Port),
+						},
+					},
+					"nat": {
+						"OVN-KUBE-EXTERNALIP": []string{
+							fmt.Sprintf("-p %s -d %s --dport %v -j DNAT --to-destination %s:%v", service.Spec.Ports[0].Protocol, externalIPv4, service.Spec.Ports[0].Port, clusterIPv4, service.Spec.Ports[0].Port),
+						},
+					},
+				}
+
+				f4 := iptV4.(*util.FakeIPTables)
+				err := f4.MatchState(expectedTables4)
+				Expect(err).NotTo(HaveOccurred())
+
+				expectedTables6 := map[string]util.FakeTable{
```

Comment:
I assume somehow go-iptables is smart enough to know to add the ipv6 rules to the ip6tables? It looks like we just add all the rules together and then hand it down to the library. I'm missing something here...

Reply:
I don't fully follow. In this case we're using two separate tables to assert that the IPv* addresses are correct. Are you saying that I should just use one `expectedTables` variable, which combines the IPv6 and IPv4 addresses needed for the assertion?

Reply:
you explained this to me about the iptables handler switching via protocol between v4 and v6. Makes sense now, thanks.